motion-pipeline/motion_extraction/audio_analysis/__init__.py

```diff
@@ -3,18 +3,19 @@
 import typing as t
 import numpy as np
 from pathlib import Path
-from .audio_tools import calculate_8beat_segments_with_midpoints, load_audio, save_audio_from_video
+from .audio_tools import calculate_8beat_segments_with_midpoints, load_audio, MusicPhrase
 from .tempo_analysis import TempoInfo, calculate_tempo_info
-from .similarity_analysis import calculate_cross_similarity
+from .similarity_analysis import calculate_cross_similarity, compute_segment_groupings
 from ..dancetree import DanceTree, DanceTreeNode
 
 @dc.dataclass
 class AudioAnalysisResult(dcj.DataClassJsonMixin):
-        duration: float
-        sample_rate: int
-        tempo_info: TempoInfo
-        eight_beat_segments: t.List[t.List[float]]
-        cross_similarity: t.List[t.List[float]]
+    duration: float
+    sample_rate: int
+    tempo_info: TempoInfo
+    musical_phrases: t.List[MusicPhrase]
```

Comment:
I replaced `eight_beat_segments` with a more generic (and intuitive) "musical_phrases" property. The logic being that we may have other ways of defining musical phrases in the future (like would be necessary in a 3/4 time signature song, for example, which might have a 6-count phrase).